src/twisted/web/test/test_newclient.py

```diff
@@ -962,6 +961,32 @@ def test_ignored1XXResponseCausesLog(self):
         self.assertEquals(event["log_format"], "Ignoring unexpected {code} response")
         self.assertEquals(event["code"], 103)
 
+    def test_headerShortEnough(self):
+        """Test that HTTP responses with header lines up to 65536 bytes long
+        are allowed."""
+        prefix = b"a: "
+        line = prefix + b"a" * (65536 - len(prefix)) + b"\r\n"
+        response = b"HTTP/1.1 200 OK\r\n" + line + b"\r\n"
+        protocol = HTTPClientParser(
+            Request(b"GET", b"/", _boringHeaders, None), lambda ign: None
+        )
+        protocol.makeConnection(StringTransport())
+        protocol.dataReceived(response)
+        self.assertEqual(protocol.transport.disconnecting, False)
+
+    def test_headerTooLong(self):
+        """Test that HTTP responses with header lines up to 65536 bytes long
+        are allowed."""
+        prefix = b"a: "
+        line = prefix + b"a" * (65536 - len(prefix) + 1) + b"\r\n"
```

Comment:
Given https://github.com/twisted/twisted/pull/12094/files/3ade9422f18248993f2921a1439649bd39440c15#r1473704052, maybe it makes sense to keep this a valid header, so that checking that the parsed header list is empty has more value (i.e. it is not empty because it was not a valid header, it is empty due to the length limit).